pandas/io/formats/format.py

```diff
@@ -295,6 +298,9 @@ def _get_footer(self) -> str:
         footer = ""
 
         if getattr(self.series.index, "freq", None) is not None:
+            assert isinstance(
+                self.series.index, (ABCDatetimeIndex, ABCPeriodIndex, ABCTimedeltaIndex)
```

Comment:
don't we have a ABCDatetimelike ?

Reply:
No, and I only found two instances of this pattern.